sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SchemaV2Writer.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming.state
+
+import java.io.StringReader
+
+import org.apache.hadoop.fs.FSDataOutputStream
+
+import org.apache.spark.sql.types.StructType
+
+class SchemaV2Writer extends SchemaWriter {
```

Comment:
> Shall we do the same with read path?

Do you mean it's better to have a factory for writer too?
At first, I considered to have it but didn't because I guessed we have only one version of writer implementation at once (now it's V2).
V1 is used only for testing so I've moved it to the test code.
Do you think it's better that V1 writer is provided by the writer factory even though it's only used for testing?